### DISCUSSION TITLE
Tests work from within VS Code, but not via the sbt console

### DISCUSSION THREAD

**Discussion by nchammas** [scalameta/metals#2541] [Q&A] [answered]
I've successfully loaded a Scala project into VS Code with Metals, and can build everything fine. I can also run tests from within the VS Code UI successfully.

<img width="152" alt="Screen Shot 2021-02-22 at 11 53 28 AM" src="https://user-images.githubusercontent.com/1039369/108741291-95e40600-7504-11eb-978a-7cb4a7e9b4a5.png">

However, if I try to run these same tests by starting `sbt` and then running something like `testOnly com...SomeTest` in the sbt console, I get this:

```
> something/testOnly com.something.*
[info] Compiling 1 Scala source to .../target/scala-2.11/classes ...
Unrecognized VM option 'CMSClassUnloadingEnabled'
Error: Could not create the Java Virtual Machine.
Error: A fatal exception has occurred. Program will exit.
[error] Could not accept connection from test agent: class java.net.SocketException: Socket closed
[error] stack trace is suppressed; run last something / Test / testOnly for the full output
```

The full exception is:

```
[error] Could not accept connection from test agent: class java.net.SocketException: Socket closed
[error] java.net.SocketException: Socket closed
[error]         at java.base/sun.nio.ch.NioSocketImpl.endAccept(NioSocketImpl.java:689)
[error]         at java.base/sun.nio.ch.NioSocketImpl.accept(NioSocketImpl.java:762)
[error]         at java.base/java.net.ServerSocket.implAccept(ServerSocket.java:684)
[error]         at java.base/java.net.ServerSocket.platformImplAccept(ServerSocket.java:650)
[error]         at java.base/java.net.ServerSocket.implAccept(ServerSocket.java:626)
[error]         at java.base/java.net.ServerSocket.implAccept(ServerSocket.java:583)
[error]         at java.base/java.net.ServerSocket.accept(ServerSocket.java:540)
[error]         at sbt.ForkTests$Acceptor$1$.run(ForkTests.scala:71)
[error]         at java.base/java.lang.Thread.run(Thread.java:832)
```

This used to work until I installed Metals, so I suspect Bloop is somehow the cause of this problem. Otherwise, what I do in an independent terminal session shouldn't have any relationship to what I'm doing in VS Code. Before trying Metals, I was able to load this project in IntelliJ and simultaneously run tests via the sbt console in a separate terminal session without issue.

What should I do to troubleshoot?

The project runs on sbt 1.3.13 and Scala 2.11.12.

**Comment by tgodzik** [comment, marked as answer]
It seems it's failing to start the forked test process due to `CMSClassUnloadingEnabled`, but Metals does not add that option. Do you know where is it coming from? It shouldn't also come from Bloop really.

**Reply by nchammas** [reply]
Oh... Interesting. That's in our build.sbt. I commented it out and can now run the tests via the sbt console.

Sorry, I feel silly. Scala newbie so I just glossed over the details of the output.

The fact that this worked for me until I tried Metals convinced me that Metals was somehow the cause, but I guess something else is going on.

**Reply by tgodzik** [reply]
No worries, I would also think it's connected, however I really have no idea why it would come up at the same time..

**Reply by nchammas** [reply]
OK, I got to the bottom of it. Compare:

```
$ java -version
openjdk version "1.8.0_282"
```

```
$ sbt
[info] welcome to sbt 1.3.13 (N/A Java 15.0.1)
```

I'm guessing that when I installed Metals, Java 15 somehow ended up on sbt's path.

After normalizing my Java environment using SDKMAN (suggested to me by my co-workers), sbt points to the expected Java version (1.8) and the original build.sbt once again works as-is.
